wrpcorrectify/options.go

```diff
@@ -0,0 +1,203 @@
+// SPDX-FileCopyrightText: 2022 Comcast Cable Communications Management, LLC
+// SPDX-License-Identifier: Apache-2.0
+
+package wrpcorrectify
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/xmidt-org/wrp-go/v3"
+)
+
+// ErrorOption returns an option that always returns the given error.
+func ErrorOption(err error) Option {
+	return OptionFunc(func(_ context.Context, m *wrp.Message) error {
+		return err
+	})
+}
+
+// Options returns a new option that applies all of the given options in order.
+func Options(opts ...Option) Option {
+	return OptionFunc(func(ctx context.Context, m *wrp.Message) error {
+		for _, opt := range opts {
+			if opt != nil {
+				if err := opt.Correctify(ctx, m); err != nil {
+					return err
+				}
+			}
+		}
+		return nil
+	})
+}
+
+// ReplaceAnySelfLocator replaces any `self:` based locator with the scheme and
+// authority of the given locator.  If the given locator is not valid, the
+// option returns an error.
+func ReplaceAnySelfLocator(me string) Option {
```

Comment:
You might be able to cut down the code repetition by have fewer or just one option that was parameterized.  For example:

```
func ReplaceAnyLocator(scheme string, me string) Option { ...}
```

You could also use struct fields via the `reflect` package to bind together a replacement value with the `wrp.Message` fields it applies to.